File: backend/internal/cluster/model/node.go
package model

import (
	corev1 "k8s.io/api/core/v1"
)

type NodeInfo struct {
	Name             string               `json:"name" label:"名称"`
	Status           string               `json:"status" label:"状态"`
	Roles            string               `json:"roles" label:"角色"`
	Version          string               `json:"version" label:"版本"`
	InternalIP       string               `json:"internal_ip" label:"内部IP"`
	ExternalIP       string               `json:"external_ip" label:"外部IP"`
	Architecture     string               `json:"architecture" label:"架构"`
	Unschedulable    bool                 `json:"unschedulable" label:"不可调度"`
	PodCount         int                  `json:"pod_count" label:"Pod数量"`
	Labels           map[string]string    `json:"labels" label:"标签"`
	Taints           []corev1.Taint       `json:"taints" label:"污点"`
	IsReady          bool                 `json:"is_ready" label:"是否就绪"`
	Addresses        []corev1.NodeAddress `json:"addresses" label:"地址"`
	CapacityCPU      string               `json:"capacity_cpu" label:"总cpu"`
	CapacityMemory   string               `json:"capacity_memory" label:"总内存"`
	AllocatableCPU   string               `json:"allocatable_cpu" label:"可用cpu"`
	AllocatableMem   string               `json:"allocatable_mem" label:"可用内存"`
	OSImage          string               `json:"os_image" label:"操作系统"`
	KernelVersion    string               `json:"kernel_version" label:"内核版本"`
	ContainerRuntime string               `json:"container_runtime" label:"容器运行时"`
	Age              string               `json:"age" label:"年龄"`
}

File: frontend/src/api/node.ts
import request from './request'
import type { AxiosResponse } from 'axios'

// ============ 类型定义 ============

export interface NodeInfo {
  name: string
  status: string
  roles: string
  version: string
  internal_ip: string
  external_ip: string
  architecture: string
  unschedulable: boolean
  pod_count: number
  labels: Record<string, string>
  taints: { key: string; value: string; effect: string }[]
  is_ready: boolean
  os_image: string
  kernel_version: string
  container_runtime: string
  creationTimestamp: string
  cpu_used: number
  cpu_total: number
  mem_used: number
  mem_total: number
  pod_total: number
}

export interface NodeDetail {
  name: string
  status: string
  roles: string
  version: string
  os: string
  kernel: string
  container_runtime: string
  architecture: string
  internal_ip: string
  external_ip: string
  hostname: string
  unschedulable: boolean
  labels: Record<string, string>
  taints: { key: string; value: string; effect: string }[]
  conditions: {
    type: string
    status: string
    reason: string
    message: string
    lastTransitionTime: string
  }[]
  capacity: Record<string, string>
  allocatable: Record<string, string>
  creationTimestamp: string
}

export interface NodeEvent {
  type: string
  reason: string
  message: string
  last_seen: string
}

export interface DrainResult {
  evicted: string[]
  skipped: string[]
  failed: string[]
}

export interface K8sPod {
  name: string
  namespace: string
  creationTimestamp: string
  status: {
    phase: string
    podIP: string
    conditions: { type: string; status: string }[]
    containerStatuses: { restartCount: number }[]
  }
}

// ============ Node API ============

export function getNodeList(params?: { clusterName?: string }): Promise<AxiosResponse<NodeInfo[]>> {
  return request.get('/k8s/cluster/nodes', { params })
}

export function getNodeDetail(params: { name: string }): Promise<AxiosResponse<NodeDetail>> {
  return request.get('/k8s/node/detail', { params })
}

export function getNodeYaml(params: { name: string }): Promise<AxiosResponse<{ yaml: string }>> {
  return request.get('/k8s/node/get-yaml', { params })
}

export function updateNodeYaml(data: { name: string; yaml: string }): Promise<AxiosResponse<null>> {
  return request.put('/k8s/node/update-yaml', data)
}

export function cordonNode(data: {
  name: string
  cordon: boolean
}): Promise<AxiosResponse<{ isCordon: boolean }>> {
  return request.put('/k8s/node/cordon', data)
}

export function updateNodeTaints(data: {
  name: string
  taints: { key: string; value: string; effect: string }[]
}): Promise<AxiosResponse<null>> {
  return request.put('/k8s/node/taints', data)
}

export function updateNodeLabels(data: {
  name: string
  labels: Record<string, string>
}): Promise<AxiosResponse<null>> {
  return request.put('/k8s/node/labels', data)
}

export function drainNode(data: {
  name: string
  ignoreDaemonSets?: boolean
  deleteLocalData?: boolean
  gracePeriod?: number
  force?: boolean
}): Promise<AxiosResponse<DrainResult>> {
  return request.put('/k8s/node/drain', data)
}

export function deleteNode(data: { name: string }): Promise<AxiosResponse<null>> {
  return request.delete('/k8s/node/delete', { data })
}

export function getNodePods(params: { name: string }): Promise<AxiosResponse<K8sPod[]>> {
  return request.get('/k8s/node/pods', { params })
}

export function getNodeEvents(params: { name: string }): Promise<AxiosResponse<NodeEvent[]>> {
  return request.get('/k8s/node/events', { params })
}
